litmus-portal/graphql-server/pkg/project/create.go

```diff
@@ -136,72 +136,31 @@ func SendInvitation(ctx context.Context, member model.MemberInput) (*model.Membe
 	if err != nil {
 		return "Unsuccessful", err
 	}
-
-	if invitation == dbSchemaProject.AcceptedInvitation {
-		return "Unsuccessful", errors.New("You are already a member of this project")
-	} else if invitation == dbSchemaProject.PendingInvitation {
-		err := dbOperationsProject.UpdateInvite(ctx, member.ProjectID, member.UserID, dbSchemaProject.AcceptedInvitation, nil)
-		if err != nil {
-			return "Unsuccessful", err
-		}
-		return "Successfull", nil
-	} else if invitation == dbSchemaProject.DeclinedInvitation {
-		return "Unsuccessful", errors.New("You have already declined the request")
-	} else if invitation == dbSchemaProject.ExitedProject {
-		return "Unsuccessful", errors.New("You are no longer a member of this project")
-	}
-
-	return "Unsuccessful", errors.New("No invitation is present to accept")
+	return "Successfull", nil
 }
 
 // DeclineInvitation :Decline an Invitaion
 func DeclineInvitation(ctx context.Context, member model.MemberInput) (string, error) {
 
-	invitation, err := getInvitation(ctx, member)
+	err := dbOperationsProject.UpdateInvite(ctx, member.ProjectID, member.UserID, dbSchemaProject.DeclinedInvitation, nil)
 	if err != nil {
 		return "Unsuccessful", err
 	}
-
-	if invitation == dbSchemaProject.AcceptedInvitation {
-		return "Unsuccessful", errors.New("You are already a member of this project")
-	} else if invitation == dbSchemaProject.PendingInvitation {
-		err := dbOperationsProject.UpdateInvite(ctx, member.ProjectID, member.UserID, dbSchemaProject.DeclinedInvitation, nil)
-		if err != nil {
-			return "Unsuccessful", err
-		}
-		return "Successfull", nil
-	} else if invitation == dbSchemaProject.DeclinedInvitation {
-		return "Unsuccessful", errors.New("You have already declined the request")
-	} else if invitation == dbSchemaProject.ExitedProject {
-		return "Unsuccessful", errors.New("You are no longer a member of this project")
-	}
-
-	return "Unsuccessful", errors.New("No invitation is present to decline")
+	return "Successfull", nil
 }
 
 //LeaveProject :Leave a Project
 func LeaveProject(ctx context.Context, member model.MemberInput) (string, error) {
 
-	invitation, err := getInvitation(ctx, member)
+	err := dbOperationsProject.UpdateInvite(ctx, member.ProjectID, member.UserID, dbSchemaProject.ExitedProject, nil)
 	if err != nil {
 		return "Unsuccessful", err
 	}
-
-	if invitation == dbSchemaProject.AcceptedInvitation {
-		err := dbOperationsProject.UpdateInvite(ctx, member.ProjectID, member.UserID, dbSchemaProject.ExitedProject, nil)
-		if err != nil {
-			return "Unsuccessful", err
-		}
-		return "Successfull", err
-	} else if invitation == dbSchemaProject.PendingInvitation || invitation == dbSchemaProject.DeclinedInvitation || invitation == dbSchemaProject.ExitedProject {
-		return "Unsuccessful", errors.New("You are not a member of this project")
-	}
-
-	return "Unsuccessful", errors.New("No Project to perform this operation on (Leave Project)")
```

Comment:
why is this not required anymore?

Reply:
These cases are already handled using RBAC. We already have a check on `invitation status` in `validate.go` package